core/templates/dev/head/services/StateRulesStatsServiceSpec.js

```diff
@@ -138,5 +138,52 @@ describe('State Rules Stats Service', function() {
       );
       expect(failureHandler).not.toHaveBeenCalled();
     });
+
+    it('should convert FractionInput into readable strings', function() {
+      var successHandler = jasmine.createSpy('success');
+      var failureHandler = jasmine.createSpy('failure');
+
+      $httpBackend.expectGET(
+        '/createhandler/state_rules_stats/7/Fraction').respond(
+        {
+          visualizations_info: [{
+            data: [
+              {
+                answer: {
+                  isNegative: false,
+                  wholeNumber: 0,
+                  numerator: 1,
+                  denominator: 2
+                },
+                frequency: 3
+              },
+              {
+                answer: {
+                  isNegative: false,
+                  wholeNumber: 0,
+                  numerator: 0,
+                  denominator: 1
+                },
+                frequency: 5
+              }]
+          }]
+        });
+
+      StateRulesStatsService.computeStateRulesStats({name: 'Fraction',
```

Comment:
nitpick: please keep sub-expressions as vertically-tight as possible.

For example, the object you create (`{name: 'Fraction', interaction: {id: 'FractionInput'}}`) can fit on a single line, so keep it on one so readers can understand it is a single "unit" at a glance.

So something like:
```javascript
      StateRulesStatsService.computeStateRulesStats(
        {name: 'Fraction', interaction: {id: 'FractionInput'}}).then(
          successHandler, failureHandler);
```